util/http.cc

```diff
@@ -171,6 +171,7 @@ static int do_post(const char* url, int connect_timeout, int read_timeout,
   if (!error) {
     logger->info("Was able to POST to {} - status code: {}", url, http_code);
   }
+  curl_slist_free_all(headers);
```

Comment:
Given the mix of C++ features/calls between the allocation and the free, do you think it would make sense to use a smart pointer for this to improve exception safety?